catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java

```diff
@@ -259,6 +257,24 @@ InputTransformerErrorHandler getInputTransformerErrorHandler() {
         }
         return this.inputTransformerErrorHandler;
     }
+
+    /**
+     * Defines and returns a {@link DynamicMetacardType} based on component Sax Event Handlers
+     * and what attributes they populate
+     *
+     * @return a DynamicMetacardType that describes the type of metacard that is created in this transformer
+     */
+    public MetacardType getMetacardType(String id) {
+        Set<AttributeDescriptor> attributeDescriptors =
+                new HashSet<>(BasicTypes.BASIC_METACARD.getAttributeDescriptors());
+
+        attributeDescriptors.addAll(eventHandlers.stream()
+                .map(SaxEventHandler::getSupportedAttributeDescriptors)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet()));
+
+        return new DynamicMetacardType(attributeDescriptors, id);
```

Comment:
Looking at `DynamicMetacardType`, I think this will actually change the metacard type name.  I am not sure this is what we want to happen.  Why can we not use `MetacardTypeImpl` here?